### PR TITLE
Upgrade Alpine, kubectl, awscli

### PR DESCRIPTION
**Pull request by nateg-gladly**
Upgrading to the current Alpine release so our `apk` commands can install the most recent and secure version of packages

Also upgrade kubectl and awscli to latest

### Testing
Will cut a release of this repo, and bump the version here: https://github.com/sagansystems/release-harness/blob/master/Dockerfile

Then run an AD build to see if it can deploy correctly

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM alpine:3.12
+FROM alpine:3.15
 
 ARG KUBE_VERSION="v1.16.10"
```
Comment (darend) [inline Dockerfile:3]: This should be 1.21.x

Comment (darend) [reply Dockerfile:3]: Can you also upgrade the aws cli to the latest on line 5

Comment (nateg-gladly) [reply Dockerfile]: done!

File: Dockerfile [hunk 2]
```diff
@@ -1,8 +1,8 @@
-FROM alpine:3.12
+FROM alpine:3.15
 
-ARG KUBE_VERSION="v1.16.10"
+ARG KUBE_VERSION="v1.23.4"
 ENV AWS_DEFAULT_REGION us-west-2
-ENV AWSCLI_VERSION 1.19.107
+ENV AWSCLI_VERSION 1.22.73
```
Comment (darend) [inline Dockerfile:5]: Note that we officially moved to v2 2 years ago https://aws.amazon.com/blogs/developer/aws-cli-v2-is-now-generally-available/

Comment (nateg-gladly) [reply Dockerfile:5]: Hmmm interesting... is v2 not used much?

The awscli release notes look like they're only releasing 1.x point releases, and are releasing them frequently: https://github.com/aws/aws-cli/blob/develop/CHANGELOG.rst

And the python package we use also is on 1.x: https://pypi.org/project/awscli/

There is this separate python package `awscliv2` but the repo only has 17 stars https://github.com/vemel/awscliv2

Comment (darend) [reply Dockerfile:5]: You need to look at the v2 branch, https://github.com/aws/aws-cli/blob/v2/CHANGELOG.rst

Comment (nateg-gladly) [reply Dockerfile:5]: Ohh got it cool. v2 doesn't need python so probably we can remove all of the python from this chain of images too. Can I do that in a separate PR to keep these atomic? Changing to v2 will mean changing our installation to this method: https://docs.aws.amazon.com/cli/latest/userguide/getting-started-version.html and testing it. Merging the PR as-is still gets us the Alpine and kubectl upgrades and an incremental awscli one and should be more drop in

Comment (darend) [reply Dockerfile:5]: Yes!

**Review by darend — APPROVED** [review]

**Pull request merged**